my-vault/internal/handlers/handlers.go

```diff
@@ -0,0 +1,181 @@
+type SecretHandler struct {
+	DB SecretsRepository
+}
+
+type CreateSecretRequest struct {
+	Name  string            `json:"name"`
+	KVMap map[string]string `json:"kv_map"`
+}
+
+type CreateSecretResponse struct {
+	Message string `json:"message"`
+	Id      string `json:"id"`
+	Name    string `json:"name"`
+}
+type GetSecretResponse struct {
+	Message string            `json:"message"`
+	Id      string            `json:"id"`
+	Name    string            `json:"name"`
+	KVMap   map[string]string `json:"kv_map"`
+}
+
+type ListSecretsResponse struct {
+	Message string          `json:"message"`
+	Secrets []domain.Secret `json:"Secrets"`
+}
+
+func GenerateUUID() string {
+	newUUID, err := uuid.NewV7()
+	if err != nil {
+		fmt.Println("Error generating UUID:", err)
+		return err.Error()
+	}
+	return newUUID.String()
+}
+
+func NewSecretResponseObject(s domain.Secret) GetSecretResponse {
+	return GetSecretResponse{
+		Message: "Retrieved secret object",
+		Id:      s.Id,
+		Name:    s.Name,
+		KVMap:   s.KVMap,
+	}
+}
+
+func NewCreateSecret(s CreateSecretRequest) domain.Secret {
+	return domain.Secret{
+		Id:    GenerateUUID(),
+		Name:  s.Name,
+		KVMap: s.KVMap,
+	}
+}
+
+func BuildListSecretsResponse(s []domain.Secret) ListSecretsResponse {
+	return ListSecretsResponse{
+		Message: "Secrets fetched successfully",
+		Secrets: s,
+	}
+}
+
+func (h *SecretHandler) HandlePostSecret(w http.ResponseWriter, r *http.Request) {
+
+	//var secret Secret
+	var secretRequestObject CreateSecretRequest
+
+	err := json.NewDecoder(r.Body).Decode(&secretRequestObject)
+	defer r.Body.Close()
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		log.Printf("ERROR: Unexpected behaviour: %v", err)
+		return
+	}
+	// build new secretObject
+	temp := NewCreateSecret(secretRequestObject)
+	//err = db.PutNewSecret(temp)
+	err = h.DB.PutNewSecret(temp)
+
+	if err != nil {
+
+		err := errors.New("unexpected internal error")
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		log.Printf("ERROR: Unexpted internal error: %v", err)
+		return
+	}
+
+	res := CreateSecretResponse{
+		Id:      temp.Id,
+		Name:    temp.Name,
+		Message: fmt.Sprintf("Successfully created secret with secretId: %s and name: %s", temp.Id, temp.Name),
+	}
+	
+	log.Printf("INFO: Secret saved----------------------------------------- %s", temp.Id)
+	w.Header().Set("Content-Type", "application/json")
+	encoder := json.NewEncoder(w)
+	err = encoder.Encode(res)
+	if err != nil {
+
+		err := errors.New("unexpected internal error")
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		log.Printf("ERROR: Unexpted internal error: %v", err)
+		return
+	}
+}
+
+func (h *SecretHandler) HandleGetSecretById(w http.ResponseWriter, r *http.Request) {
+	//curl  -X POST -H "Content-Type: application/json" http://localhost:9000/secrets/234
+	secretID := r.PathValue("secret_id")
+	secretEntry, err := h.DB.GetScretsById(secretID)
+	
+	if err.Error() == ErrNotFound.Error(){
```

Comment:
Errors can be compared directly, no need to use the string representation like this
```suggestion
	if err == storage.ErrNotFound {
```
Though it is often better to handle both this error and any [_wrapped_ errors](https://go.dev/blog/go1.13-errors) using `errors.Is`:
```go
    if errors.Is(err, storage.ErrNotFound) {
```